### PR TITLE
Make it clear that being able to restore root is a failure

### PR DESCRIPTION
**Pull request by emersion**
Related to #3024

### CODE REVIEW

File: sway/main.c [hunk 1]
```diff
@@ -206,7 +206,7 @@ static void drop_permissions(void) {
 		}
 	}
 	if (setuid(0) != -1) {
-		wlr_log(WLR_ERROR, "Root privileges can be restored.");
+		wlr_log(WLR_ERROR, "Root privileges can be restored. Refusing to start.");
```
Comment (Alphare) [inline sway/main.c]: Wouldn't it be clearer to say something like "
```suggestion
		wlr_log(WLR_ERROR, "Dropping root privileges failed. Refusing to start.");
``` ?

File: sway/main.c [hunk 2]
```diff
@@ -194,21 +194,23 @@ static void log_kernel(void) {
...
-			exit(EXIT_FAILURE);
+			wlr_log(WLR_ERROR, "Unable to drop root, refusing to start");
+			return false;
 		}
 	}
 	if (setuid(0) != -1) {
-		wlr_log(WLR_ERROR, "Root privileges can be restored.");
-		exit(EXIT_FAILURE);
+		wlr_log(WLR_ERROR, "Unable to drop root (root privileges can be "
+			"restored, and shouldn't), refusing to start");
```
Comment (SirCmpwn) [inline sway/main.c]: "Unable to drop root - we shouldn't be able to restore it after setuid - refusing to start"

Comment (emersion) [reply sway/main.c]: Done

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**